### PR TITLE
Bluetooth: audio: ascs: Remove redundant sessions pool

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This removes redundant bt_ascs structure pool that held the list of ases per connection. The memory usage can be reduced by moving the conn reference to bt_ascs_ase structure. This change reduces flash and RAM memory usage.

### CODE REVIEW

**Comment by MariuszSkamra** [comment]
#AutoPTS run zephyr ASCS

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/zephyrproject-rtos/zephyr/pull/57231#issuecomment-1522881124 after 08:56:22.

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
Failed tests:
ASCS    ASCS/SR/ACP/BV-07-C             BTP ERROR

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -51,49 +51,19 @@ BUILD_ASSERT(CONFIG_BT_ASCS_MAX_ACTIVE_ASES <= MAX(MAX_ASES_SESSIONS,
...
 	struct bt_bap_ep ep;
 	const struct bt_gatt_attr *attr;
-	sys_snode_t node;
 	struct k_work_delayable disconnect_work;
-};
-
-struct bt_ascs {
-	struct bt_conn *conn;
-	sys_slist_t ases;
-};
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:65]: It does feel a bit weird that we don't have an ASCS struct, as current (before this PR) structure with an ASCS per connection pointer, and then 0 or more ASEs per ASCS feels like a more logical structure, and better adheres to the way that the ASCS specification is written. 

However, based on the changes in this PR, I agree that it seems that there is no technical advantage of the above structure, and I do not see any alternatives that is using less resources that what this PR implements.

**Review by Thalley — APPROVED** [review]
Changes somewhat disconnects the implementation from the specification with the removal of the `bt_ascs` structure, but the proposed changes makes sense from a resource-oriented perspective. 

@MariuszSkamra it would be interesting to see how much memory we actually save by the changes; could you try to run the `ram_report` for e.g. the shell or the HAP HA sample before and after this PR, and post the result here?

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c:65]: TL;DR
I've checked hap_ha sample and the result is as follows
It saves 40bytes of RAM and 312bytes of ROM.
Bear in mind that after the changes, the ASCS size is independent of `CONFIG_BT_MAX_CONN`.

`CONFIG_BT_MAX_CONN=1`:
Before:
image
RAM
image
ROM
image

After:
image
RAM
image
ROM
image

Currently, setting `CONFIG_BT_MAX_CONN=2` affects the memory usage as follows:
image
image
As you can see, increasing the number of connections by one increased RAM usage by 12bytes more

**Review by szymon-czapracki — APPROVED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/audio/ascs.c:65]: Very nice; thanks for the report :)

**Pull request merged**